src/towncrier/_builder.py

```diff
@@ -69,25 +100,21 @@ def parse_newfragment_basename(
 # Also returns a list of the paths that the fragments were taken from.
 def find_fragments(
     base_directory: str,
-    sections: Mapping[str, str],
-    fragment_directory: str | None,
-    frag_type_names: Iterable[str],
-    orphan_prefix: str | None = None,
+    config: Config,
```

Comment:
I think this change is on.

I am not sure what we consider public or private API.

I can do something like this, which might suggest that this is a public API.

```
 $ python 
Python 3.11.9 (main, Apr  6 2024, 17:59:24) [GCC 11.4.0] on linux
>>> import towncrier
>>> from towncrier.check import find_fragments
```

this is one reason, why I think that we should always prefix private code with an underline.

It make it easier to check with code is public and which is not.


My suggestion is to rename this to `_find_fragments` to make it clear that it should not be used, outside of `towncrier`

Reply:
If this is the case, there is a *lot* of methods that we should rename. The public API in my opinion, is anything we have documented. It shouldn't matter if it's underscored or not -- if you're using an undocumented method of a project, I don't see why you should assume it should continue working.

Maybe we need to have an official agreement on this and document the policy?

Reply:
Opened #605 for discussion.

Reply:
I agree that there are a lot of methods... but at least we should try no to add more :)

Feel free to leave this as it is.


I will add more info on #605